homeassistant/components/sensor/worxlandroid.py

```diff
@@ -121,43 +118,51 @@ def async_update(self):
             if self.sensor == 'battery':
                 self._state = data['perc_batt']
 
+            # sensor state
+            elif self.sensor == 'state':
```

Comment:
Looks like we can use the same dict here if it maps sensor type to a list of values, where the items are name and state key.